frappe/__init__.py

```diff
@@ -1480,7 +1480,10 @@ def get_value(*args, **kwargs):
 
 def as_json(obj, indent=1):
 	from frappe.utils.response import json_handler
-	return json.dumps(obj, indent=indent, sort_keys=True, default=json_handler, separators=(',', ': '))
+	try:
+		return json.dumps(obj, indent=indent, sort_keys=True, default=json_handler, separators=(',', ': '))
+	except TypeError:
+		return json.dumps(obj, indent=indent, default=json_handler, separators=(',', ': '))
```

Comment:
@surajshetty3416 `as_json`'s return value feels inconsistent. If we're returning values differently depending on the data implicitly, we might as well stop sorting altogether. We could also do without the extra overhead IMO 😄

Reply:
While fixing the issue the thought was to avoid breaking the expected behavior (which is to have sorted keys) from `as_json` and also handle the very rare case (object has "None" as a key) with a minor caveat of responding without sorted keys in case of comparison failure.

I don't think we should suddenly remove this implicit sorting now (the assumption is that it might break somewhere).

---

The other fix I can think of is to remove any "None" key in the object before passing it to `json.dumps` because anyway it generates an invalid JSON as mentioned [here](https://github.com/frappe/frappe/issues/14292#issuecomment-927330924).

Reply:
I too feel we should pop any `None` keys. 

---

Also, I'm not sure how useful sort_keys even is or why the client would expect a JSON sorted by keys. From what I'm seeing its behaviour might seem unpredictable (`A > _ > a > z`). Let's remove it for now, from `develop`?

<img width="478" alt="Screenshot 2022-03-14 at 1 53 10 PM" src="https://user-images.githubusercontent.com/36654812/158132892-a61ab2a7-6021-4cf2-b4e4-c26cceacfd0e.png">